[DOC] Update how Search query builder works (#122891)
grafana/grafana: docs/sources/shared/datasources/tempo-search-traceql.md

For example, **Span Name** and **Service Name** are string fields so the comparison operators are equals (`=`), not equal (`!=`), matches regular expressions (`=~`), or doesn't match regular expression (`!~`).
**Duration** is a duration field type and uses range selections (`>`, `>=`, `<`, `<=`).

You can select multiple values for the same filter.
The generated query depends on the operator you choose:

- **Equals (`=`) or other non-regex operators**: Each value becomes a separate condition joined with `||` inside parentheses. For the not-equals (`!=`) operator, conditions are joined with `&&` instead.

> - **Regex match (`=~`) or regex not match (`!~`)**: Values are concatenated with `|` inside a single quoted string.

Author: ```suggestion
- Equals (`=`) or other non-regular expression operators: Each value becomes a separate condition joined with `||` inside parentheses. For the not-equals (`!=`) operator, conditions are joined with `&&` instead.
- Regular expressions match (`=~`) or not match (`!~`): Values are concatenated with `|` inside a single quoted string.
```

This capability only applies to fields with drop-down value selection.

For example, if you select **Span Name** with the `=` operator and choose both `get` and `log_results_cache` as values, the generated query uses separate conditions:

```traceql
{ name="get" || name="log_results_cache" }
```

> If you select the regex operator `=~` instead, the values are combined into a single regex pattern:

Author: ```suggestion
If you select the regular expression operator `=~` instead, the values are combined into a single regex pattern:
```

> If you select the regular expression operator `=~` instead, the values are combined into a single regex pattern:

Author: ```suggestion
If you select the regular expression operator `=~` instead, the values are combined into a single regular expression pattern:
```
